### PR TITLE
Use base platform matrix for CI tests

### PR DESCRIPTION
**Pull request by benbp**
This PR updates our CI build definition to use the shared [platform matrix config](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/stages/platform-matrix.json), along with using the dynamically generated matrix approach to tests.

The CI and live tests are now driven off the same base test matrix, though for CI I tuned the sparseness a little bit to cover all OS*JavaTestVersion combinations (as it is currently).

The overall change is minimal, but looks larger since I moved the ci test job out into its own file (since it gets included as a template multiple times by the generate job, which we don't want to do for build/analyze), as well as rename it.

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  - name: CloudConfig
+    type: object
+    default: {}
+  - name: DependsOn
+    type: string
+    default: ''
+  - name: UsePlatformContainer
+    type: boolean
+    default: false
+
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:37]: nit: 3 empty lines

**Review by weshaggard — APPROVED** [review]
Looks reasonable once tested.

**Review by alzimmermsft — APPROVED** [review]
Could you run AppConfiguration, Cosmos, Spring, and Resource Manager CIs as additional validation. All have unique scenarios.

File: eng/pipelines/templates/stages/cosmos-sdk-client.yml [hunk 2]
```diff
@@ -28,62 +28,29 @@ parameters:
...
         ServiceDirectory: ${{parameters.ServiceDirectory}}
         Artifacts: ${{ parameters.Artifacts }}
         AdditionalModules: ${{ parameters.AdditionalModules }}
         SDKType: ${{parameters.SDKType}}
-        TestMatrix:
-          Linux - Java 8:
-            Pool: azsdk-pool-mms-ubuntu-1804-general
-            OSVmImage: MMSUbuntu18.04
-            JavaTestVersion: '1.8'
-            RunTitle: 'Linux on Java 1.8'
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/cosmos-sdk-client.yml:42]: Looks like RunTitle was missed in the matrix translation.

**Pull request merged**